Sources/LiveKit/Core/SignalClient.swift

```diff
@@ -217,6 +210,9 @@ actor SignalClient: Loggable {
         await _addTrackCompleters.reset()
         await _requestQueue.clear()
         await _responseQueue.clear()
+
+        self.disconnectError = LiveKitError.from(error: disconnectError)
+        connectionState = .disconnected
```

Comment:
why move this? does it change anything?

Reply:
I think it changes the timing of `didUpdateConnectionState:` delegate firing.